docker-compose.yml

```diff
@@ -1,16 +1,16 @@
-citusmaster:
-    build: .
-    environment:
-        CITUS_MASTER: true
-    ports: ["5432:5432"]
-    links:
-        - citus
-        - citus2
+version: '2'
 
-citus:
+services:
+  master:
+    container_name: 'citus_master'
     build: .
-    ports: ["5001:5432"]
-
-citus2:
+    ports: ['5432:5432']
+    labels: ['com.citusdata.role=Master']
+  worker:
     build: .
-    ports: ["5002:5432"]
+    labels: ['com.citusdata.role=Worker']
+  config:
+    container_name: 'citus_config'
+    image: 'citusdata/workerlist-gen:0.9.0'
+    volumes: ['/var/run/docker.sock:/tmp/docker.sock']
```

Comment:
Yes.

The `citusdata/workerlist-gen` container uses a program called `docker-gen` to listen to docker events and regenerate certain files after certain events. See the top-level description on this pull request for links. From `docker-gen`'s documentation:

> There are three common ways to run docker-gen:
> * on the host
> * bundled in a container with another application
> * separate standalone containers

I'm going with option three, which is to run `docker-gen` in a container with access to (a) the docker socket (to listen to the API), and (b) a shared volume with the Citus worker list file (so it can update the worker lists).

`docker-gen` has an option that tells it to send a `SIGHUP` to another container after a template has been regenerated. This is how we can signal Citus to reload config after a `docker-compose scale` action.